### ISSUE TITLE
[feat] Add `no-unnecessary-use-ref`

### ISSUE THREAD

**Issue by SukkaW** [Rel1cx/eslint-react#1394]
### Problem Description

Something similar to https://github.com/Rel1cx/eslint-react/issues/1278 , this time for `useRef`.

This code should be considered invalid:

```tsx
const rafIdRef = useRef<number | null>(null);

useEffect(() => {
  radIdRef.current = requestAnimationFrame();
  return () => {
    if (rafIdRef.current != null) { cancleAnimationFrame(rafId.current) }
  }
}, []);
```

This is because the ref `rafIdRef` is only used in one `useEffect`, thus the variable should be kept scoped and local, not in a ref:

```tsx
useEffect(() => {
  let radId = requestAnimationFrame();
  return () => {
    if (radId != null) { cancleAnimationFrame(rafId) }
  }
}, []);
```

The same thing applies to callback refs (basically, they are just layout effects).

However, if the `useRef` is being used in more than one useEffect and/or more than one ref callback, then this `useRef` usage should be considered valid. 

### Alternative Solutions

N/A

### Feature Details

Maybe we could call this rule: no-useless-use-ref

### Examples

See above.

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users
